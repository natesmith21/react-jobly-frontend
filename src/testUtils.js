import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasApplied = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasApplied }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };