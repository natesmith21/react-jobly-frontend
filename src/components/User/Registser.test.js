import React from "react";
import { render } from "@testing-library/react";
import Register from "./Register";
import { UserProvider } from "../../testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
      <UserProvider>
        <Register />
      </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});