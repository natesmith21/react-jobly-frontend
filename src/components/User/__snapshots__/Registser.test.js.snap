// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1>
      Register
    </h1>
    <form
      class=""
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="email"
        >
          Username
        </label>
        <input
          class="form-control"
          id="username"
          name="username"
          placeholder="username"
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="password"
        >
          Password
        </label>
        <input
          class="form-control"
          id="password"
          name="password"
          placeholder="password"
          type="password"
          value=""
        />
        <label
          class="form-label"
          for="firstName"
        >
          First Name
        </label>
        <input
          class="form-control"
          id="firstName"
          name="firstName"
          placeholder="First Name"
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="lasntName"
        >
          First Name
        </label>
        <input
          class="form-control"
          id="lastName"
          name="lastName"
          placeholder="Last Name"
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="email"
        >
          Email
        </label>
        <input
          class="form-control"
          id="email"
          name="email"
          placeholder="email"
          type="email"
          value=""
        />
      </div>
      <button
        class="btn btn-secondary"
      >
        Sign Me Up!
      </button>
    </form>
  </section>
</DocumentFragment>
`;
